src/rgw/h3/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+# this is a separate cmake project so Ceph can build it as an ExternalProject
+
+cmake_minimum_required(VERSION 3.16)
+project(h3)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/")
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+find_package(OpenSSL REQUIRED)
+find_package(Quiche 0.16.0 REQUIRED)
+find_package(Boost 1.81 REQUIRED COMPONENTS system)
```

Comment:
to satisfy this, had to change:
![image](https://user-images.githubusercontent.com/31659604/220617286-d5f1661b-156f-4b2b-831e-741377427420.png)
but then there are link errors:
```
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg10'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg9'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg8'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg7'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg6'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg5'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg4'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg3'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg2'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(ConfUtils.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg1'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg10'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg9'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg8'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg7'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg6'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg5'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg4'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg3'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg2'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
/usr/local/libexec/gold/ld: error: lib/libcommon.a(MonCap.cc.o): multiple definition of 'boost::phoenix::placeholders::uarg1'
/usr/local/libexec/gold/ld: lib/libosd.a(OSDCap.cc.o): previous definition here
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.

```

Reply:
oops, i had been experimenting with this on top of the wip-coro-after-reef feature branch where we updated to 1.81. i switched it back to 1.79